### DISCUSSION TITLE
Compiling for Windows - Confusing OpenSSL Section

### DISCUSSION THREAD

**Discussion by nihilisticFish** [FreeRDP/FreeRDP#7719] [Q&A]
I got pretty far but now on the OpenSSL section for building in Windows I can't resolve the perl line.

I am assuming that the build directory is the freerdp source directory and the source directory is the location of where I compiled OpenSSL but the /Configure is confusing me. I get the error that says there is no such file or directory which is totally true. I don't see any file or folder in either location called Configure.

I have tried switching the build/source directories with the same result.

Please advise. Thank you.

**Comment by akallabeth** [comment]
Please do not mix compiling `OpenSSL` with compiling `FreeRDP`.
They have nothing to do with each other. `FreeRDP` just uses a (installed) `OpenSSL`.
Check the `OpenSSL` documentation on how to compile that properly, then start compiling `FreeRDP`

**Comment by nihilisticFish** [comment]
I already compiled OpenSSL I am talking about the building process for FreeRDP. Under the "Building Section" under "OpenSSL" the documentation is unclear specifically where the line that says:
'perl <path to source>/Configure shared --prefix=<path to install directory> VC-WIN32 (or `VC-WIN64A` for 64 bit builds)'

I tried both paths and neither works, what is the specific  "path to source" and where is /Configure?

**Reply by akallabeth** [reply]
You are talking about https://github.com/FreeRDP/FreeRDP/wiki/Compilation ?
Please read the section title ;)
1. You're quoting the section for compiling and installing `OpenSSL`
2. Fill in the proper paths in that line : `perl <path to source>/Configure shared --prefix=<path to install directory> VC-WIN32` and not just skip them (and wonder why nothing works)

**Reply by nihilisticFish** [reply]
I am quoting the path for your convenience since you were unclear on where I was. My confusion is that I have tried placing the directory paths for source and receive an error. There is no /Configure in the OpenSSL directory.

C:\freerdp-2.6.1_TMP>perl "C:\Program Files\OpenSSL"/Configure shared --prefix="C:\FreeRDP" VC-WIN64A
Can't open perl script "C:\Program Files\OpenSSL/Configure": No such file or directory

**Reply by akallabeth** [reply]
@nihilisticFish You´re still confusing `compiling OpenSSL` with `installing OpenSSL`
If you already installed `OpenSSL` binaries you do not need to `Compile OpenSSL`...
And not finding source files in binary installations is not really surprising.
As reference:
1. This is the source code for OpenSSL: https://github.com/openssl/openssl and there you also find the build system with the [Configure](https://github.com/openssl/openssl/blob/master/Configure) file.

**Reply by nihilisticFish** [reply]
I see, that does make more sense now that I look at that again. For some reason I thought that the "Building" section was somehow setting up those binaries to be used by FreeRDP.

Thank you for the patience, I was definitely confused.
